src/screens/buy/exolix/use-topup-update.hook.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { refreshExolixExchangeDataAction } from '../../../store/exolix/exolix-actions';
+import { useExolixExchangeData } from '../../../store/exolix/exolix-selectors';
+import { isDefined } from '../../../utils/is-defined';
+
+const useTopUpUpdate = (setIsError: (isError: boolean) => void) => {
+  const exchangeData = useExolixExchangeData();
+  const dispatch = useDispatch();
+  const setExchangeData = useCallback(
+    (id: string) => {
+      dispatch(refreshExolixExchangeDataAction(id));
+    },
+    [dispatch]
+  );
+  const isAlive = useRef(false);
+
+  useEffect(() => {
+    let timeoutId = setTimeout(async function repeat() {
```

Comment:
why fn declaration?

Reply:
look below for another setTimeout calling after execution current timer: `timeoutId = setTimeout(repeat, 3000);`

Reply:
look below for another setTimeout calling after execution current timer: timeoutId = setTimeout(repeat, 3000);